File: Data Structures & Algorithms/climbing-stairs/submission-1.py
class Solution:
    def climbStairs(self, n: int) -> int:
        res = []
        for i in range(1, n+1):
            if i == 1:
                res.append(1)
            elif i ==2:
                res.append(2)
            else:
                print(res)
                res.append(res[i-2] + res[i-3])
        return res[len(res)-1]

File: Data Structures & Algorithms/combination-target-sum/submission-1.java
class Solution {
    Set<List<Integer>> res;
    private void compute(int[] nums, int sum, int index, int target, List<Integer> curr) {
        if (sum == target) {
            res.add(List.copyOf(curr));
            return;
        }
        if (index >= nums.length || sum > target) {
            return;
        }
        curr.add(nums[index]);
        compute(nums, sum+nums[index], index, target, curr);
        curr.remove(curr.size()-1);
        compute(nums, sum, index+1, target, curr);

    }
    public List<List<Integer>> combinationSum(int[] nums, int target) {
        res = new HashSet<>();
        List<Integer> curr = new ArrayList<>();
        compute(nums, 0, 0, target, curr);
        return new ArrayList<>(res);

    }
}
